### PR TITLE
Leo tally circuit

### PR DESCRIPTION
**Pull request by hexLancel**
Basic tally circuit in Leo

### CODE REVIEW

File: circuits/tally/src/poseidon.leo [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+            state = Poseidon::mix(state);
+        }
+        return state[0];
+    }
+
+    function ark(state: [field;3], it: u32 ) -> [field;3]{
+        // state[i].add_assign(&c[it + i]);
+        const c_constants: [field;195] = 
+        [
+            1278823093112533518033294982858707370905937792805078104831550399751562346033,
```
Comment (mationorato) [inline circuits/tally/src/poseidon.leo:45]: Are we sure is it safe to have this fields in the repo? should we check it with errol?

**Review by mationorato — COMMENTED** [review]
mauro where is the cuadratic complexity? so i can we aware of.

File: circuits/tally/src/poseidon.leo (hunk 1 above)
Comment (hexLancel) [reply circuits/tally/src/poseidon.leo:45]: Yes we are

Comment (hexLancel) [reply circuits/tally/src/poseidon.leo:45]: The constants are public in every Poseidon implementation, they are not secrets

Comment (mationorato) [reply circuits/tally/src/poseidon.leo:45]: great, i will marge it

**Pull request merged**